teps/0047-pipeline-task-display-name.md

```diff
@@ -0,0 +1,160 @@
+title: Pipeline Task Display Name
+creation-date: '2021-01-28'
+last-updated: '2021-02-09'
+authors:
+- '@itewk'
+---
+
+# TEP-0047: Pipeline Task Display Name
+---
+
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+A task in a Pipeline is currently represented in the UI (as observed in OpenShift 4.6)
+using a field (`name`) that is meant to be machine readable, not human readable.
+There should be the addition of a new optional field (`displayName`) to describe
+Pipeline tasks that are human readable so that any tool that renders Tekton Pipeline
+can do so using a "human readable" field.
+
+## Motivation
+
+The end user of a Tekton pipeline can vary including but not limited to
+application developer, to security professional, to compliance officer,
+service reliability engineer, product manager, and beyond. The farther
+you move away from the application developer the more important human
+understanding of the CI/CD process is important.
+
+Currently if a product manager or auditor was to look at a Tekton Pipeline run rendered in UI
+that would be presented with a bunch of truncated machine readable task names, no spaces,
+no special characters, no capitalization.
+
+Compare this to other tools in this space, Jenkins, GitLab CI, GitHub Actions, etc.
+Their renderings of their workflows are meant to be human consumable. Their task names have
+the option to be rendered in a human consumable format with use of any characters and fully
+rendered for context.
+
+### Goals
+
+* Add a `Pipeline.spec.tasks.*.displayName` (optional) (or similar) field that allows any text and
+  if provided will be useable by UIs rendering Tekton Piplines rather then the
+  machine readable `Pipeline.spec.tasks.*.name` field.
+* Add a `Pipeline.spec.tasks.*.description` (optional) (or similar) field that allows for a
+  long form description of a task in the context of a `Pipeline` to provide additional
+  human context to the intent of a task in a Pipeline.
+* Add a `Pipeline.spec.displayName` (optional) (or similar) field that allows any text
+  and if provided will be useable for UIs rendering Tekton `Pipeline` rather then the
+  machine readable `Pipeline.name` field.
```

Comment:
If we add it here, why not adding it in `Task` ? Not that one `Task`, in the catalog, so far we are using labels for the `displayName` (`tekton.dev/displayName: "git clone"`).

https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md#example-of-a-resource-from-catalog

I think we should be consistent on all *definition* types (Pipeline, Task, …).